drivers/syslog/syslog_putc.c

```diff
@@ -55,48 +55,57 @@
 
 int syslog_putc(int ch)
 {
-  DEBUGASSERT(g_syslog_channel != NULL);
+  int i;
 
-  /* Is this an attempt to do SYSLOG output from an interrupt handler? */
-
-  if (up_interrupt_context() || sched_idletask())
+  for (i = 0; i < CONFIG_SYSLOG_MAX_CHANNELS; i++)
     {
-#if defined(CONFIG_SYSLOG_INTBUFFER)
-      if (up_interrupt_context())
-        {
-          /* Buffer the character in the interrupt buffer.
-           *  The interrupt buffer will be flushed before the next normal,
-           * non-interrupt SYSLOG output.
-           */
+      if (g_syslog_channel[i] == NULL)
+        break;
+
+      /* Is this an attempt to do SYSLOG output from an interrupt handler? */
 
-          return syslog_add_intbuffer(ch);
+      if (up_interrupt_context() || sched_idletask())
+        {
+#if defined(CONFIG_SYSLOG_INTBUFFER)
+          if (up_interrupt_context())
+            {
+              /* Buffer the character in the interrupt buffer.
+               * The interrupt buffer will be flushed before the next
+               * normal,non-interrupt SYSLOG output.
+               */
+
+              return syslog_add_intbuffer(ch);
```

Comment:
The same ch may put into the interrupt buffer duplicately.

Reply:
I don't understand the problem, the loop exited there.  
But nevertheless, is it fixed now?